app/init.go

```diff
@@ -0,0 +1,135 @@
+	"github.com/iov-one/weave/x/multisig"
+	"github.com/iov-one/weave/x/validators"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+// GenInitOptions will produce some basic options for one rich
+// account, to use for dev mode
+func GenInitOptions(args []string) (json.RawMessage, error) {
+	// Your coins ticker code
+	code := "PRJCT"
+	if len(args) > 0 {
+		code = args[0]
+	}
+
+	var addr string
+	if len(args) > 1 {
+		addr = args[1]
+	} else {
+		// if no address provided, auto-generate one
+		// and print out a recovery phrase
+		bz, phrase, err := GenerateCoinKey()
+		if err != nil {
+			return nil, err
+		}
+		addr = bz.String()
+		fmt.Println(phrase)
+	}
+
+	type (
+		dict  map[string]interface{}
+		array []interface{}
+	)
+	collectorAddr, err := hex.DecodeString("3b11c732b8fc1f09beb34031302fe2ab347c5c14")
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot hex decode collector address")
+	}
+	return json.Marshal(dict{
+		"cash": array{
+			dict{
+				"address": addr,
+				"coins": array{
+					dict{
+						"whole":  123456789,
+						"ticker": code,
+					},
+				},
+			},
+		},
+		"currencies": array{
+			dict{
+				"ticker": "PRJCT",
+				"name":   "Your projects native token",
+			},
+		},
+
+		"conf": dict{
+			"cash": cash.Configuration{
+				CollectorAddress: collectorAddr,
+				MinimalFee:       coin.Coin{Whole: 0}, // no fee
+			},
+			"migration": dict{
+				"admin": addr,
+			},
+		},
+		"initialize_schema": []dict{
+			{"pkg": "custom", "ver": 1},
+			{"pkg": "cash", "ver": 1},
+			{"pkg": "sigs", "ver": 1},
+			{"pkg": "multisig", "ver": 1},
+			{"pkg": "utils", "ver": 1},
+			{"pkg": "migration", "ver": 1},
+			{"pkg": "validators", "ver": 1},
+		},
+	})
+}
+
+// GenerateApp is used to create a stub for server/start.go command
+func GenerateApp(options *server.Options) (abci.Application, error) {
+	// db goes in a subdir, but "" -> "" for memdb
+	var dbPath string
+	if options.Home != "" {
+		dbPath = filepath.Join(options.Home, "abci.db")
+	}
+
+	stack := Stack(nil, options.MinFee)
+	application, err := Application("project", stack, TxDecoder, dbPath, options.Debug)
+	if err != nil {
+		return nil, err
+	}
+
+	return DecorateApp(application, options.Logger), nil
+}
+
+// DecorateApp adds initializers and Logger to an Application
+func DecorateApp(application app.BaseApp, logger log.Logger) app.BaseApp {
+	application.WithInit(app.ChainInitializers(
+		&cash.Initializer{},
+		&multisig.Initializer{},
+		&currency.Initializer{},
+		&migration.Initializer{},
+		&validators.Initializer{},
+	))
+	application.WithLogger(logger)
+	return application
+}
+
+// GenerateCoinKey returns the address of a public key,
+// along with the secret phrase to recover the private key.
+// You can give coins to this address and return the recovery
+// phrase to the user to access them.
+func GenerateCoinKey() (weave.Address, string, error) {
+	// TODO implement BIP39 recovery phrases
```

Comment:
I think a TODO does not belong in a good codebase. Either implement it or leave a good comment explaining *why* below functionality works in this certain way and what was the reason for choosing this implementation.

Leaving TODO like this will give an impression that the code is not finished and lacking, therefore cannot be trusted. No one will address this comment as well.